ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryUrlSigningService.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.frontend;
+
+import com.github.ambry.rest.RestMethod;
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.rest.RestServiceErrorCode;
+import com.github.ambry.rest.RestServiceException;
+import com.github.ambry.rest.RestUtils;
+import com.github.ambry.utils.Time;
+import java.util.Map;
+
+
+/**
+ * Default implementation of {@link UrlSigningService} that currently only converts all POST headers into query
+ * parameters and does not actually reliably sign the URL.
+ * <p/>
+ * A real implementation needs to encode enough information for manipulations to be discovered.
```

Comment:
Maybe we could add a todo and a link to GH issue

Reply:
Good idea. Added the link. Didn't add a TODO either.